### PR TITLE
Stop rendering empty paragraph when there is no question help text

### PR DESCRIPTION
**Pull request by jcma-google**
### Description

Remove padding beneath the question help text unless question help text is present.

## Release notes

The title of the pull request will be used as the default release notes description. If more detail is needed to communicate to partners the scope of the PR's changes, use this release notes section.

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [ ] Added an additional reviewer from outside your organization as FYI (if the primary reviewer is in the same organization as you)
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)
- [x] Extended the README / documentation, if necessary

### Issue(s) this completes

Fixes #<issue_number>; Fixes #<issue_number>...

### CODE REVIEW

File: browser-test/src/applicant/questions/text.test.ts [hunk 1]
```diff
@@ -162,6 +162,33 @@ test.describe(
       })
     })
 
+    test.describe('single text question without help text', () => {
```
[the remark this answers is not in the document]
Comment (jcma-google) [reply browser-test/src/applicant/questions/text.test.ts:165]: The test above verifies that the help text is shown when present.

File: browser-test/src/applicant/questions/text.test.ts [hunk 2]
```diff
@@ -162,6 +162,33 @@ test.describe(
...
+      const programName = 'Test program for single text without help text'
+
+      test.beforeEach(async ({page, adminQuestions, adminPrograms}) => {
```
[the remark this answers is not in the document]
Comment (jcma-google) [reply browser-test/src/applicant/questions/text.test.ts:168]: I originally wanted to group this test with the describe block above, however, the block above "single text question" looks like it's explicitly validating single text question state. I can add a test that adds a text question without help text but it would be against the spirit of the describe block. I added a separate describe block as a compromise but if you feel that the test would be better grouped with the first describe block even if it defies the mandate (we can't have a single text question with help text and without help text).

File: browser-test/src/applicant/questions/text.test.ts [hunk 3]
```diff
@@ -162,6 +162,33 @@ test.describe(
...
+        await adminPrograms.addAndPublishProgramWithQuestions(
+          ['text-q'],
+          programName,
+        )
+
+        await logout(page)
+        await disableFeatureFlag(page, 'north_star_applicant_ui')
+      })
+
+      test('validate screenshot', async ({page, applicantQuestions}) => {
```
[the remark this answers is not in the document]
Comment (jcma-google) [reply browser-test/src/applicant/questions/text.test.ts:185]: Validating that the help text eleemnt is empty. I was thinking about validating that the class "mb-4" is not present but that feels very fragile (mb-4 is not a unique classname).

File: browser-test/src/applicant/questions/text.test.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (jcma-google) [reply browser-test/src/applicant/questions/text.test.ts]: Looks like after sheldon reformatted the tests there is a clearer pattern of separating the different testing conditions with describe, I'm going to resolve this conversation for now (looks like nick is at a conference this week) and will sync up with him via chat after he is back to see if he would prefer to reformat differently.

File: server/app/views/questiontypes/ApplicantQuestionRendererImpl.java [hunk 4]
```diff
@@ -73,12 +73,12 @@ public final DivTag render(ApplicantQuestionRendererParams params) {
                     .with(
                         TextFormatter.formatTextWithAriaLabel(
                             applicantQuestion.getQuestionHelpText(),
-                            /* preserveEmptyLines= */ true,
+                            /* preserveEmptyLines= */ false,
```
Comment (devin-civi) [inline server/app/views/questiontypes/ApplicantQuestionRendererImpl.java:76]: Presumably this will have an effect on help text that contains empty lines in it? 

(For example if I have some help text like like:
"Please enter from the following options:

* foo
* bar"

will the empty line after "Please enter..." be removed?)

Not sure how much we care if so...

File: browser-test/image_snapshots/end_to_end_enumerators-End-862d8-merator-elements-in-preview-retry1/end-to-end-enumerators-test-enumerator-type-set.png
Comment (devin-civi) [inline browser-test/image_snapshots/end_to_end_enumerators-End-862d8-merator-elements-in-preview-retry1/end-to-end-enumerators-test-enumerator-type-set.png:1]: Where do these screenshots come from? Seems unrelated to this?

File: browser-test/image_snapshots/end_to_end_enumerators-End-862d8-merator-elements-in-preview/end-to-end-enumerators-test-enumerator-type-set.png
Comment (devin-civi) [inline browser-test/image_snapshots/end_to_end_enumerators-End-862d8-merator-elements-in-preview/end-to-end-enumerators-test-enumerator-type-set.png:1]: Same, doesn't seem we should be adding screenshots

File: server/app/assets/javascripts/preview.ts [hunk 5]
```diff
@@ -252,6 +252,7 @@ class PreviewController {
         PreviewController.QUESTION_HELP_TEXT_SELECTOR,
       )
       if (contentParent) {
+        contentParent.classList.add('mb-4')
```
[the remark this answers is not in the document]
Comment (jcma-google) [reply server/app/assets/javascripts/preview.ts:255]: Yup, it adds margin when help text is present. Otherwise the help text is too close to the other divs. We originally added mb-4 server-side to the help-text container but it was rendering empty space when there wasn't help-text.

File: browser-test/image_snapshots/end_to_end_enumerators-End-862d8-merator-elements-in-preview-retry1/end-to-end-enumerators-test-enumerator-type-set.png
Comment (jcma-google) [reply browser-test/image_snapshots/end_to_end_enumerators-End-862d8-merator-elements-in-preview-retry1/end-to-end-enumerators-test-enumerator-type-set.png:1]: Mhm I keep onremoving these screenshots and they keep on sneaking back into my commits. Going to try it one more time.

File: server/app/views/questiontypes/ApplicantQuestionRendererImpl.java (hunk 4 above)
Comment (jcma-google) [reply server/app/views/questiontypes/ApplicantQuestionRendererImpl.java:76]: The empty space is unnecessary because of the margin added on  mb-4. If there is no string then we don't want to render a line. We can add spacing with the margin.

Comment (devin-civi) [reply server/app/views/questiontypes/ApplicantQuestionRendererImpl.java:76]: To clarify: I'm asking if an admin writes some help text which contains empty lines in the middle of their help text, will this remove them? Or does it only affect lines at the end/completely empty help text?

Comment (jcma-google) [reply server/app/views/questiontypes/ApplicantQuestionRendererImpl.java]: Woops, accidentally clicked resolve, still getting the hang of this UI. 

This removes empty lines throughout the help text. Thanks for the catch! Keeping prserveEmptyLines to true would preserve the existing functionality and not interfere with the fix.

**Comment by gwendolyngoetz** [comment]
Related https://github.com/civiform/civiform/issues/8537
Fixed by https://github.com/civiform/civiform/pull/8657
